sympy/polys/sqfreetools.py

```diff
@@ -381,13 +619,35 @@ def dmp_sqf_list(f, u, K, all=False):
     >>> R.dmp_sqf_list(f, all=True)
     (1, [(1, 1), (x + y, 2), (x, 3)])
 
+    Explanation
+    ===========
+
+    Uses Yun's algorithm for univariate polynomials from [Yun76]_ recrusively.
+    The multivariate polynomial is treated as a univariate polynomial in its
+    leading variable. Then Yun's algorithm computes the square-free
+    factorization of the primitive and the content is factored recursively.
+
+    It would be better to use a dedicated algorithm for multivariate
+    polynomials instead.
```

Comment:
Fair if you don't have particular candidate algorithms in mind, but if you do have some references, might be good to provide them as a starting point for future developers.

Reply:
If I had found a good reference describing a simple algorithm for this then I would have implemented it.

The original Yun76 paper did describe a multivariate algorithm as well:
https://dl.acm.org/doi/pdf/10.1145/800205.806320

However that algorithm is more complicated because it depends on the EZGCD algorithm which is not implemented in sympy. I am not sure that it is in general worth implementing EZGCD because I believe it is superseded by other GCD algorithms.

Yun also refers to previous papers that have multivariate algorithms involving Hensel lifting that I assume are also more complicated.

It seems common in textbooks, theses etc to describe Yun's univariate algorithm but not mention the multivariate case.

More recent papers generally often describe the case of multivariate polynomials in positive characteristic e.g.

> On square-free factorization of multivariate over a finite field
Laurent Bernardin 1997

Bernardin compares the algorithm for multivariate polynomials in positive characteristic with Yun's multivariate algorithm in characteristic zero. Bernardin's algorithm is claimed to be simpler and have other advantages. It looks a lot like Yun's univariate algorithm but it is for positive characteristic.

I can't seem to find a simple description of the multivariate characteristic zero case. I think possibly the reason for that is because it is "trivial" somehow like maybe Bernardin's algorithm can be trivially adapted to characteristic zero. It doesn't seem trivial to me though...

Bernardin's algorithm requires computing a $p^l$-th root of a polynomial for maximal $l$ where $p$ is the characteristic. I'm not sure what to do with that in the characteristic zero case. Maybe we can just skip that step?

Reply:
I'm happy to keep thinking about this one, but don't have any good ideas right now.

Anyway, this thread can serve as a good starting point for future work.